### ISSUE TITLE
Save stats in coolers

### ISSUE THREAD

**Issue by Phlya** [mirnylab/distiller-nf#90]
Hi, I was thinking it would be cool if stats about the experiment could be saved directly into the cooler files too (in metadata/info?). This would make this information be always associated with the cooler file, no need to calculate cis/trans since the information is already there, and would in general make comparing multiple experiments easier since everything is easily accessible from python.

**Comment by golobor** [comment]
hmmm, that's a good idea. @nvictus, any thoughts?
Transferring pairs-level stats into coolers may be a bit clumsy, since it'll add one more command-line flag to   type and one more file to track. Ideally, stats should be stored in the header of a pairs file and then get transferred to .cool files. But with .pairs files currently being a single text file, the implementation will be slow. I think storing stats in the header will get more natural once we switch to the binary storage format for pairs.

The current alternative is to expand calculation of statistics in cooler. Specifically, we can add cis/trans, scalings, chromosome-vs-chromosome counts, etc. @Phlya, @nvictus  what do you think?

**Comment by Phlya** [comment]
I agree that if stats are stored in a separate file it becomes a little complicated, and if they were in the header it would be much simpler.

The alternative idea sounds good to me - and these stats could also then be calculated at creation of the cooler file (optionally?) and stored in it.
